chain/chain/src/store_validator/validate.rs

```diff
@@ -181,43 +200,27 @@ pub(crate) fn block_chunks_exist(
 }
 
 pub(crate) fn block_height_cmp_tail(
-    sv: &mut StoreValidator,
-    _key: &[u8],
-    value: &[u8],
-) -> Result<(), ErrorMessage> {
-    let tail = unwrap_or_err!(
-        sv.store.get_ser::<BlockHeight>(ColBlockMisc, TAIL_KEY),
-        "Can't get Tail from storage"
-    )
-    .unwrap_or(sv.config.genesis_height);
-    let block = unwrap_or_err!(Block::try_from_slice(value), "Can't deserialize Block");
-    if block.header.inner_lite.height < tail
-        && block.header.inner_lite.height != sv.config.genesis_height
-    {
-        sv.block_heights_less_tail.push(block.hash());
-    }
-    Ok(())
-}
-
-pub(crate) fn block_height_cmp_tail_count(
     sv: &mut StoreValidator,
     _key: &[u8],
     _value: &[u8],
 ) -> Result<(), ErrorMessage> {
-    if sv.block_heights_less_tail.len() < 2 {
+    sv.inner.functions_executed.insert(get_parent_function_name!());
+    assert!(sv.inner.is_block_height_cmp_tail_prepared);
+    if sv.inner.block_heights_less_tail.len() < 2 {
```

Comment:
what's this magic?

Reply:
After `5254f85c70` `TAIL` is not pointing to the earliest Block anymore. There may be up to two blocks below `TAIL`: Genesis and the first alive Block in the Chain. As I don't count Genesis, `block_heights_less_tail` should be less than 2 in consistent Storage.

Reply:
> first alive Block in the Chain

Why?

Reply:
because `TAIL` is not pointing to the block
could you please be more specific?